cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/OAuthHandler.java

```diff
@@ -106,7 +107,9 @@ public void getAuthURL(HttpServiceRequest request, HttpServiceResponder responde
   @PUT
   @Path(API_VERSION + "/oauth/provider/{provider}")
   public void putOAuthProvider(HttpServiceRequest request, HttpServiceResponder responder,
-                               @PathParam("provider") String oauthProvider) {
+                               @PathParam("provider") String oauthProvider,
+                               @QueryParam("skip_client_credentials") @DefaultValue("false")
```

Comment:
I don't like the name much. It does not tell what exactly is skipped (we are skipping the write). May be something like reuse_client_credentials or write_client_credentials?